src/components/menu/menu-item/menu-item.component.js

```diff
@@ -1,33 +1,211 @@
-import React from 'react';
+import React, {
+  useRef, useState, useEffect, useCallback
+} from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import StyledMenuItemWrapper from './menu-item.style';
 import { StyledSubmenu, StyledSubmenuItem, StyledSubmenuTitle } from '../submenu-block/submenu.style';
 import OptionHelper from '../../../utils/helpers/options-helper';
 import Link from '../../link';
+import Events from '../../../utils/helpers/events';
 
-const MenuItem = ({
+const MenuItem = React.forwardRef(({
   submenu,
   children,
   href,
   to,
   menuType,
   onClick,
   target,
-  submenuDirection,
+  submenuDirection = 'right',
   icon,
   selected,
-  routerLink
-}) => {
+  routerLink,
+  isFirstElement,
+  handleKeyDown,
+  isOpen,
+  setIsOpenSubmenu,
+  menuItemIndex
+}, ref) => {
+  const [didOpenByArrowUp, setOpenByArrowUp] = useState(false);
+  const [didOpenByMouseEnter, setOpenByMouseEnter] = useState(false);
+  const [didOpenByClick, setOpenByClick] = useState(false);
+  const submenuItemsRefs = submenu && useRef(React.Children.map(children, child => child.ref || React.createRef()));
+  const ifRef = () => !submenu && { ref };
```

Comment:
Why is the ref conditional based on submenu? The `ref` prop should be accessible by the user and should be consistent.

e.g. 

```js
const myRef = useRef();
<Menu><MenuItem ref={myRef}>Option One</MenuItem></Menu>
```

Reply:
+1, looks super confusing, if the conditional is necessary here just write it directly on the component
`<StyledMenuItemWrapper  { ...(!submenu && { ref } />`

Reply:
I moved the conditional to the component without additional function `ifRef`, I need to have the conditional here because if there is a `submenu` prop we need to push the ref deeper to another one and also we don't want to put `ref` in a `div` element